ui/src/app/components/with-leo-cookie.spec.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+
+import { withLeoCookie } from 'app/components/with-leo-cookie';
+import {
+  leoProxyApi,
+  registerApiClient as leoRegisterApiClient,
+} from 'app/services/notebooks-swagger-fetch-clients';
+import { authStore, notificationStore } from 'app/utils/stores';
+import { ProxyApi } from 'notebooks-generated/fetch';
+
+import { waitOneTickAndUpdate } from 'testing/react-test-helpers';
+import { LeoProxyApiStub } from 'testing/stubs/leo-proxy-api-stub';
+
+const createWrapper = () => {
+  const Component = () => <div />;
+  const WrappedComponent = withLeoCookie(Component);
+  return mount(<WrappedComponent />);
+};
+
+describe(withLeoCookie.name, () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+
+    const leoProxyApiStub = new LeoProxyApiStub();
+    leoRegisterApiClient(ProxyApi, leoProxyApiStub);
+
+    authStore.set({
+      authLoaded: true,
+      isSignedIn: true,
+      auth: {
+        // @ts-ignore
+        settings: {
+          accessTokenExpiringNotificationTimeInSeconds: 500,
+        },
+      },
+    });
+    notificationStore.set(null);
+  });
+
+  it('should render the children once the cookie loads', async () => {
```

Comment:
`with-leo-cookie` _does_ show the child component if there's an error, actually. Otherwise, after clicking out of the modal, there will be a completely blank screen:

<img width="2559" alt="image" src="https://github.com/all-of-us/workbench/assets/31020403/f5e03aa3-c982-4760-8ffd-628e3c3adc92">

I think this is a limitation of our current top-level error handling. It would be nice to see an error page if calls to config, profile, etc. fail.

Reply:
I'm open to changing this. However, as is, I think having three tests is OK because rendering children and showing errors are unrelated